lib/puppet/pops/loader/module_loaders.rb

```diff
@@ -33,7 +33,7 @@ def self.cached_loader_from(parent_loader, loaders)
       NAMESPACE_WILDCARD,
       Puppet[:libdir],
       'cached_puppet_lib',
-      [:func_4x, :datatype]
+      [:func_4x, :func_3x, :datatype]
```

Comment:
is the order important here?

Reply:
Yes, the order is significant, and it is correct in this PR since 4x must come before 3x.

Reply:
This is specified and important as we want 4x implementations to shadow 3x implementations of the same function.

Reply:
Ah in that case it would be good to assert that in the test. Currently the test just [checks that the 3 different types are included](https://github.com/puppetlabs/puppet/pull/7402/files#diff-05b9cd31b2c52f0ca24b4aa9f8b9f302R105-R107). @glennsarti or @hlindberg can you update that?